### PR TITLE
fix(compilation): always uses the content of the ast to compile the ast

### LINKED ISSUE #878: Launcher.compile() does not compile the AST changes
Launcher.compile() only compiles the content of the original file...
The interest of this method is thus very limited.

### PR DESCRIPTION
**Pull request by tdurieux**
fixes #878

### CODE REVIEW

File: src/test/java/spoon/test/compilation/CompilationTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+						this.value = m.invoke(aClass.newInstance());
+					} catch (Exception e) {
+						Assert.fail(e.getMessage());
+					}
+					notify();
+				}
+			}
+		}
+
+		MethodRunner methodRunner = new MethodRunner();
```
Comment (tdurieux) [inline src/test/java/spoon/test/compilation/CompilationTest.java]: I do not need a asynchronous call but a need a thread the change the classloader.
And this is the only way that I know to wait the end of the execution of a thread.

Comment (monperrus) [inline src/test/java/spoon/test/compilation/CompilationTest.java]: Without thread: new URLClassLoader(urls).loadClass(...).newInstance()

Comment (tdurieux) [inline src/test/java/spoon/test/compilation/CompilationTest.java]: Done

File: src/main/java/spoon/support/compiler/jdt/CompilationUnitWrapper.java [hunk 2]
```diff
@@ -43,30 +44,14 @@
 			return jdtCompiler.loadedContent.get(s);
```
Comment (monperrus) [inline src/main/java/spoon/support/compiler/jdt/CompilationUnitWrapper.java]: does this cache mean that one cannot modify - compile and then modify again and compile again?

Comment (tdurieux) [inline src/main/java/spoon/support/compiler/jdt/CompilationUnitWrapper.java]: No, it does not.
The cache is cleared before each compilation (see JDTBatchCompiler constructor)

**Comment by monperrus** [comment]
any idea about the exception in the build log: https://travis-ci.org/INRIA/spoon/builds/165035633#L3829?

**Comment by tdurieux** [comment]
The cache was not correctly handled

**Pull request merged**